### PR TITLE
`azurerm_site_recovery_replicated_vm`: handle casing issue & update validate func

### LINKED ISSUE #22075: azurerm_site_recovery_replicated_vm id case is inconsistent
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.59.0

### Affected Resource(s)/Data Source(s)

azurerm_site_recovery_replicated_vm

### Terraform Configuration Files

```hcl
resource "azurerm_site_recovery_replicated_vm" "vm-replication" {
  name                                      = "vm-replicated"
  resource_group_name                       = azurerm_resource_group.secondary.name
  recovery_vault_name                       = azurerm_recovery_services_vault.secondary.name
  source_recovery_fabric_name               = azurerm_site_recovery_fabric.primary.name
  source_vm_id                              = azurerm_windows_virtual_machine.vmrep.id
  recovery_replication_policy_id            = azurerm_site_recovery_replication_policy.policy.id
  source_recovery_protection_container_name = azurerm_site_recovery_protection_container.primary.name

  target_resource_group_id                = lower(azurerm_resource_group.secondary.id)
  target_recovery_fabric_id               = lower(azurerm_site_recovery_fabric.secondary.id)
  target_recovery_protection_container_id = lower(azurerm_site_recovery_protection_container.secondary.id)

  managed_disk {
    disk_id                    = lower(data.azurerm_managed_disk.vm_osdisk.id)
    staging_storage_account_id = azurerm_storage_account.primary.id
    target_resource_group_id   = azurerm_resource_group.secondary.id
    target_disk_type           = "Premium_LRS"
    target_replica_disk_type   = "Premium_LRS"
  }

  managed_disk {
    disk_id                    = lower(azurerm_managed_disk.vmrep_managed_disk_1.id)
    staging_storage_account_id = azurerm_storage_account.primary.id
    target_resource_group_id   = azurerm_resource_group.secondary.id
    target_disk_type           = "Premium_LRS"
    target_replica_disk_type   = "Premium_LRS"
  }

  managed_disk {
    disk_id                    = lower(azurerm_managed_disk.vmrep_managed_disk_2.id)
    staging_storage_account_id = azurerm_storage_account.primary.id
    target_resource_group_id   = azurerm_resource_group.secondary.id
    target_disk_type           = "Premium_LRS"
    target_replica_disk_type   = "Premium_LRS"
  }

  network_interface {
    source_network_interface_id = azurerm_network_interface.vmrep.id
    target_subnet_name          = azurerm_subnet.secondary.name
    target_static_ip            = "10.9.9.5"
  }

  depends_on = [
    azurerm_site_recovery_protection_container_mapping.container-mapping,
    azurerm_site_recovery_network_mapping.network-mapping,
  ]
}
```

### Debug Output/Panic Output

```shell
Change detected on ID because of casing conflict
```

### Expected Behaviour

During following plan, no change should be seen on the resource.

### Actual Behaviour

ID are detected as changed because of casing of 'subscription'.
```hcl
 # azurerm_site_recovery_replicated_vm.vm-replication 
~ resource "azurerm_site_recovery_replicated_vm" "vm-replication" {
        name                                      = "vm-replicated"
      ~ recovery_replication_policy_id            = "/Subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-secondary/providers/Microsoft.RecoveryServices/vaults/rsv-testbackup-gba-secondary/replicationPolicies/policy" -> "/subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-secondary/providers/Microsoft.RecoveryServices/vaults/rsv-testbackup-gba-secondary/replicationPolicies/policy"
      ~ source_vm_id                              = "/subscriptions/<sub-id>/resourcegroups/rg-testbackup-gba-primary/providers/microsoft.compute/virtualmachines/vmrep" -> "/subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-primary/providers/Microsoft.Compute/virtualMachines/vmrep"
      ~ target_network_id                         = "/subscriptions/<sub-id>/resourcegroups/rg-testbackup-gba-secondary/providers/microsoft.network/virtualnetworks/vnet-testbackup-gba-secondary" -> (known after apply)
      ~ target_recovery_fabric_id                 = "/Subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-secondary/providers/Microsoft.RecoveryServices/vaults/rsv-testbackup-gba-secondary/replicationFabrics/asr-a2a-default-westeurope" -> "/subscriptions/<sub-id>/resourcegroups/rg-testbackup-gba-secondary/providers/microsoft.recoveryservices/vaults/rsv-testbackup-gba-secondary/replicationfabrics/asr-a2a-default-westeurope"
      ~ target_recovery_protection_container_id   = "/Subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-secondary/providers/Microsoft.RecoveryServices/vaults/rsv-testbackup-gba-secondary/replicationFabrics/asr-a2a-default-westeurope/replicationProtectionContainers/secondary-protection-container" -> "/subscriptions/<sub-id>/resourcegroups/rg-testbackup-gba-secondary/providers/microsoft.recoveryservices/vaults/rsv-testbackup-gba-secondary/replicationfabrics/asr-a2a-default-westeurope/replicationprotectioncontainers/secondary-protection-container"
      ~ target_resource_group_id                  = "/subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-secondary" -> "/subscriptions/<sub-id>/resourcegroups/rg-testbackup-gba-secondary"
        # (6 unchanged attributes hidden)
    }
```

If I force casing with a replace, for example on 'recovery_replication_policy_id', I get the error:
```hcl
│ Error: Can not parse "recovery_replication_policy_id" as a resource id: No subscription ID found in: "Subscriptions/<sub-id>/resourceGroups/rg-testbackup-gba-secondary/providers/Microsoft.RecoveryServices/vaults/rsv-testbackup-gba-secondary/replicationPolicies/policy"
```

### Steps to Reproduce

1. terraform apply

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #22075

test
---
the failed ones are also failing on `main` branch so I assume they do not block this PR
Screenshot 2023-06-15 at 10 31 50

### CODE REVIEW

File: internal/services/recoveryservices/site_recovery_replicated_vm_resource.go [hunk 1]
```diff
@@ -230,11 +242,12 @@ func resourceSiteRecoveryReplicatedVM() *pluginsdk.Resource {
...
 							}, false),
 						},
+
 						"target_disk_encryption_set_id": {
 							Type:             pluginsdk.TypeString,
 							Optional:         true,
 							ForceNew:         true,
-							ValidateFunc:     azure.ValidateResourceID,
+							ValidateFunc:     diskencryptionsets.ValidateDiskEncryptionSetID,
 							DiffSuppressFunc: suppress.CaseDifference,
```
Comment (tombuildsstuff) [inline internal/services/recoveryservices/site_recovery_replicated_vm_resource.go:251]:
```suggestion
```

File: internal/services/recoveryservices/site_recovery_replicated_vm_resource.go [hunk 2]
```diff
@@ -678,24 +697,52 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
...
-		d.Set("target_recovery_protection_container_id", prop.RecoveryContainerId)
+		if prop.RecoveryFabricId != nil {
+			d.Set("target_recovery_fabric_id", handleAzureSdkForGoBug2824(*prop.RecoveryFabricId))
+		}
+		if prop.PolicyId != nil {
+			d.Set("recovery_replication_policy_id", handleAzureSdkForGoBug2824(*prop.PolicyId))
+		}
+		if prop.RecoveryContainerId != nil {
+			d.Set("target_recovery_protection_container_id", handleAzureSdkForGoBug2824(*prop.RecoveryContainerId))
+		}
```
Comment (tombuildsstuff) [inline internal/services/recoveryservices/site_recovery_replicated_vm_resource.go:708]: can we use the `ParseXXXInsensitively` functions for this instead

File: internal/services/recoveryservices/site_recovery_replicated_vm_resource.go [hunk 3]
```diff
@@ -678,24 +697,52 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
...
+			}
+			if a2aDetails.RecoveryBootDiagStorageAccountId != nil {
+				d.Set("target_boot_diagnostic_storage_account_id", handleAzureSdkForGoBug2824(*a2aDetails.RecoveryBootDiagStorageAccountId))
+			}
+			if a2aDetails.RecoveryCapacityReservationGroupId != nil {
+				d.Set("target_capacity_reservation_group_id", handleAzureSdkForGoBug2824(*a2aDetails.RecoveryCapacityReservationGroupId))
+			}
+			if a2aDetails.RecoveryVirtualMachineScaleSetId != nil {
+				d.Set("target_virtual_machine_scale_set_id", handleAzureSdkForGoBug2824(*a2aDetails.RecoveryVirtualMachineScaleSetId))
+			}
```
Comment (tombuildsstuff) [inline internal/services/recoveryservices/site_recovery_replicated_vm_resource.go:741]: (same here)

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @ziyeqf 

Thanks for this PR - can we update this to use the `ParseXXXInsensitively` methods instead?

Thanks!

File: internal/services/recoveryservices/site_recovery_replicated_vm_resource.go [hunk 4]
```diff
@@ -678,24 +700,131 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
...
 
 	if prop := model.Properties; prop != nil {
-		d.Set("target_recovery_fabric_id", prop.RecoveryFabricId)
-		d.Set("recovery_replication_policy_id", prop.PolicyId)
-		d.Set("target_recovery_protection_container_id", prop.RecoveryContainerId)
+		recoveryFabricId := ""
+		if fabricId := pointer.From(prop.RecoveryFabricId); fabricId != "" {
+			parsedFabricId, err := replicationfabrics.ParseReplicationFabricIDInsensitively(fabricId)
+			if err != nil {
+				return fmt.Errorf("parsing recovery_fabric_id %s: %+v", fabricId, err)
```
Comment (stephybun) [inline internal/services/recoveryservices/site_recovery_replicated_vm_resource.go:707]: The ID parsers already return detailed error messages so can we simplify all the error messages to 
```suggestion
				return err
```

**Review by stephybun — CHANGES_REQUESTED** [review]

**Comment by prashantshetage** [comment]
I am facing the same issue with Terrafom v1.5.0 and azurerm v.3.61.0

**Review by stephybun — CHANGES_REQUESTED** [review]
Tests are showing up a build failure:
```
# github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices
./site_recovery_replicated_vm_resource.go:766:48: undefined: networkParse.VirtualNetworkIDInsensitively
./site_recovery_replicated_vm_resource.go:776:45: undefined: networkParse.VirtualNetworkIDInsensitively
Process exited with code 1
```

**Comment by ziyeqf** [comment]
> Tests are showing up a build failure:
> 
> ```
> # github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices
> ./site_recovery_replicated_vm_resource.go:766:48: undefined: networkParse.VirtualNetworkIDInsensitively
> ./site_recovery_replicated_vm_resource.go:776:45: undefined: networkParse.VirtualNetworkIDInsensitively
> Process exited with code 1
> ```

The VirtualNetworkId function was re-located to `commonids` package on `main` branch... just updated

**Comment by ziyeqf** [comment]
> Tests are showing up a build failure:
> 
> ```
> # github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices
> ./site_recovery_replicated_vm_resource.go:766:48: undefined: networkParse.VirtualNetworkIDInsensitively
> ./site_recovery_replicated_vm_resource.go:776:45: undefined: networkParse.VirtualNetworkIDInsensitively
> Process exited with code 1
> ```

The network parse functions were re-located to `commonids` package on `main` branch... just updated code and triggered a test on TC

**Comment by stephybun** [comment]
@ziyeqf can you resolve the conflicts please?

**Review by stephybun — APPROVED** [review]
Thanks @ziyeqf LGTM 🍓

**Pull request merged**